File: Segundo_Cuatri/PDOO/ruby/Irrgarten/Weapon.rb
#encoding: UTF-8
module Weapon
    class Weapon
        def initialize(power,uses)
            @power = power
            @uses = uses
        end

        public def attack
            power = @power 
            unless @uses > 0 
                power = 0 
                @uses -= 1
            end
            return power
        end
        
        def to_s
            return "W[" + @power + "," + @uses + "]*"
        end
    end
end

File: Segundo_Cuatri/PDOO/java/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author airam
 */
public class Weapon {
    private float power;
    private int uses;
    
    Weapon(float strength, int durability){
        power = strength;
        uses = durability;
    }
    
    public float attack(){  
        float atk_val = 0;
        if(uses > 0){
            atk_val = power;
            uses--;
        }
        return atk_val;
    }
    
    public String toString(){
        return "W[" + power +"," + uses + "]";
    }
    
    public boolean discard(){
        return true;
    }
}

